### PR TITLE
Adds a link to create pull request from compare page

### LINKED ISSUE #189: [FR] Add the ability to create a PR from the compare page
Bitbucket has the convenient but hidden feature 'Compare' which can be found at https://bitbucket.org/{USER}/{REPO}/branches/compare . Or by clicking the '+' in a repo and clicking the 'Compare branches or tags' button.

Its a pleasant diff page but I would like a CTA which allows me to take the two branches I am comparing an 'Create a PR' where the feature and target branches are set from the branches I was looking at.

### PR DESCRIPTION
**Pull request by deepakrb**
*   [x] I tested the changes in this pull request myself
*   [x] I added Automated Tests
*   [x] I updated the README.md, with pictures if necessary

Thought I would take a look at this. It starts off with a new link 'Create Pull Request' (much like the one found in the branches page) and looks like this:

screen shot 2018-03-22 at 23 09 04

When clicked it'll show:

screen shot 2018-03-22 at 23 10 11

I took a look at testing, it would have been nicer to mock the `isComparing` method rather than pass its source and destination through.

Closes #189 
---

### CODE REVIEW

**Review by drexler — COMMENTED** [review]
I pulled and tested this locally. Works as a charm. 👍 . However, i noticed if one navigates to the /branches/compare url and then clicks on the compare button, the page needs an extra refresh before the pull request link can be displayed.

**Comment by deepakrb** [comment]
@drexler cheers for looking into it 👍 . You're right I'll post a fix

File: src/main.js [hunk 1]
```diff
@@ -60,6 +62,8 @@ function init(config) {
         }
     } else if (isCommit()) {
         codeReviewFeatures(config);
+    } else if (isComparing()) {
+        mergeCompare(config);
```
Comment (reyronald) [inline src/main.js]: Do you need to pass this `config` object ?

**Review by reyronald — COMMENTED** [review]
Remember to add a checkbox in the `options.html` to enable/disable this feature! And turn it on by default in `background.js`

File: src/main.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ import pullrequestCommitAmount from './pullrequest-commit-amount';
 import insertPullrequestTemplate from './pullrequest-template';
 import addSidebarCounters from './sidebar-counters';
 import syntaxHighlight from './syntax-highlight';
+import mergeCompare from './merge-compare';
```
Comment (reyronald) [inline src/main.js]: I'm not sure about this name :S, I have a tough time relating what the feature does to the name `merge-compare`, can you come up with something better?

**Review by reyronald — CHANGES_REQUESTED** [review]
Remember to include an entry in the CHANGELOG :D

File: src/main.js (hunk 1 above)
Comment (deepakrb) [reply src/main.js]: Nope, i'll remove it

File: src/main.js (hunk 2 above)
Comment (deepakrb) [reply src/main.js]: `compare-page-pull-request`?

**Comment by deepakrb** [comment]
I'll make those changes and squash and get back to this

**Comment by deepakrb** [comment]
@drexler @reyronald I've opened up a PR here https://github.com/refined-bitbucket/refined-bitbucket/pull/191 which should fix all these issues

**Comment by drexler** [comment]
@deepakrb i'll take a look tonight. Thx for the effort on this.

**Pull request closed without merging**